test/extensions/key_value/file_based/key_value_store_test.cc

```diff
@@ -119,9 +119,9 @@ TEST_F(KeyValueStoreTest, HandleBadFile) {
 
 #ifndef WIN32
 TEST_F(KeyValueStoreTest, HandleInvalidFile) {
-  filename_ = "/foo";
+  filename_ = "/path/that/does/not/exist/foo";
```

Comment:
If we're going to harden this, I wonder if it would make more sense to TestEnvironment::temporaryPath("foo") to guarantee that the path does not exist?

Reply:
The problem is that the created file (/tmp/xxx) might be writable, so the flush will succeed.

Reply:
The doc for this function says: `Prefix a given path with the private writable test temporary directory`.

So, if I understand it correctly it gives us a private path (writable) + the argument. This test relies on flushing the store to fail. For this, it gives a "bad" filename (`/foo`) to the store. The issue here is that this filename is not too bad. If `/` is writable, the flush will succeed (we don't want that).

Reply:
So TestEnvironment::temporaryDirectory()[1] "Obtain a private writable temporary directory." If you look at the implementation it's also empty. 

1. https://github.com/envoyproxy/envoy/blob/main/test/test_common/environment.h#L94

Reply:
So TestEnvironment::temporaryDirectory()[1] "Obtain a private writable temporary directory." If you look at the implementation it's also empty. TestEnvironment::temporaryPath() merely appends the argument to that newly created directory. So I'm proposing we use "<tempdir>/<subdir>/<filename>" as the name of the file. TestEnvironment::temporaryPath("foo") will give us "<tempdir>/<subdir>", with "foo" as <subdir> and temporaryDirectory() as <tempdir>. We'd need to append the filename as well.

Does that make sense?

1. https://github.com/envoyproxy/envoy/blob/main/test/test_common/environment.h#L94
2. https://github.com/envoyproxy/envoy/blob/main/test/test_common/environment.h#L101

Reply:
Argh! Github messed with this comment because of markdown. *sigh* Trying again:
```
 So I'm proposing we use "<tempdir>/<subdir>/<filename>" as the name of the file. TestEnvironment::temporaryPath("foo") will give us "<tempdir>/<subdir>", with "foo" as <subdir> and temporaryDirectory() as <tempdir>. We'd need to append the filename as well.
 ```

Reply:
ok, I've updated the patch to use the same pattern found in other files.